### ISSUE TITLE
Add a description of the listen method to NestFastifyApplication

### ISSUE THREAD

**Issue by grin0c** [nestjs/nest#3786]
## Describe the solution you'd like

I want to handle the fastify error when starting the app.
```typescript
await app.listen(port, host, (err: Error) => {
    if (err) {
      logger.error(err)
      process.exit(1)
    }

    logger.info(`Application is listening on port ${host}:${port}.`, 'Main')
  })
```

If you write this now, typescript throws an error, so in INestApplication, the listen method is defined like this:
```typescript
listen(port: number | string, callback?: () => void): Promise<any>;
listen(port: number | string, hostname: string, callback?: () => void): Promise<any>;
```
I suggest taking the annotation from https://github.com/fastify/fastify/blob/master/fastify.d.ts#L529-L537 and adding it to NestFastifyApplication

**Comment by kamilmysliwiec** [comment]
Looks good. Would you like to create a PR for this?

**Comment by grin0c** [comment]
I will do

**Comment by kamilmysliwiec** [comment]
Fixed in 6.11.0
